bindings/rust/s2n-tls-tokio/src/config.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use s2n_tls::raw::{config, connection::Connection, enums::Mode, error::Error};
+
+/// Produces new, un-negotiated connections.
+pub trait Config {
+    fn create(&self, mode: Mode) -> Result<Connection, Error>;
+}
+
+/// Produces new, un-negotiated connections with the given Config set.
+impl Config for config::Config {
+    fn create(&self, mode: Mode) -> Result<Connection, Error> {
+        let mut conn = Connection::new(mode);
+        conn.set_config(self.clone())?;
+        Ok(conn)
+    }
+}
+
+/// Produces new, un-negotiated connections.
+/// Useful for ad-hoc customization of connection creation.
```

Comment:
what's the use case for this?

Reply:
Probably mostly testing. It's a lighter weight alternative to actually creating a structure and implementing Builder for simple cases.